Add V2EX Viewer Extension

daikaiyu opened the pull request:
## Description

A better extension for viewing V2EX topics and replies. Support two kinds of topic details display style, support to add proxy for access V2EX.

## Screencast

### Thumbnail view
<img width="1000" alt="v2ex-viewer-1" src="https://user-images.githubusercontent.com/78251524/188321317-0c245391-6639-46a4-9d4e-8ae3c867c282.png">

### Detailed view [V2EX Style]
<img width="1000" alt="v2ex-viewer-2" src="https://user-images.githubusercontent.com/78251524/188321322-fd709e28-2763-4e4c-a716-8c1147ec5a17.png">

### Detailed view [Raycast Style]
<img width="1000" alt="v2ex-viewer-3" src="https://user-images.githubusercontent.com/78251524/188321514-397039a8-420d-4214-b092-a8e18ff5d7a2.png">


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/v2ex-viewer/src/components/TopicListItem.tsx:
```diff
@@ -0,0 +1,59 @@
+        !isShowingDetail
+          ? [
+              {
+                tooltip: "主题回复数",
```
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

pernielsentikaer commented on extensions/v2ex-viewer/src/components/TopicListItemDetail.tsx:
```diff
@@ -0,0 +1,46 @@
+      metadata={
+        topicDetailStyle === TopicDetailStyle.Raycast && (
+          <List.Item.Detail.Metadata>
+            <List.Item.Detail.Metadata.Label title="主题相关信息" />
+            <List.Item.Detail.Metadata.Label title="标题" text={topic.title} />
+            <List.Item.Detail.Metadata.Label title="节点" icon={node.avatar_mini} text={node.title} />
+            <List.Item.Detail.Metadata.Label title="发布者" icon={member.avatar_mini} text={member.username} />
+            <List.Item.Detail.Metadata.Label title="发布时间" text={getUnixFromNow(topic.created)} />
```
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

pernielsentikaer commented on extensions/v2ex-viewer/src/utils/markdown.ts:
```diff
@@ -0,0 +1,42 @@
+  const topicTitle = `${Heading(1, topic.title)}`;
+  const topicMember = `${Bold(Link(topic.member.username, topic.member.url))} · ${getUnixFromNow(topic.created)} · ${
+    topic.replies
+  } 条回复`;
```
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪

I left a few minor corrections for you, can you look into this?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

daikaiyu commented:
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I left a few minor corrections for you, can you look into this?
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

Hello! I have removed localized content and added new command. Looking forward to your review and suggestions again.

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
